### DISCUSSION TITLE
RabbitMQ Scaler help

### DISCUSSION THREAD

**Discussion by janavenkat** [kedacore/keda#5975] [General]
Hi Team,

We tried Keda with [RabbitMQ](https://keda.sh/docs/2.14/scalers/rabbitmq-queue/) scaler while ago. We noticed that scaler requests the rabbitMQ cluster for each single queue. So what's the reason for getting each queue metrics in every time instead of getting all at once? 

Reason for asking, I have cluster that has connection limits and a lot of queues.

**Comment by JorTurFer** [comment]
We do that because each scaler is independent of others. Although you could have multiple queues in the same cluster, the cluster can be shared by different teams within the org, and each time could have different permissions. If we aggregate all the information, it could require a privileged connection or a really complex logic to manage the connections with different credentials, etc
